include/cute/atom/copy_traits_xe.hpp

```diff
@@ -0,0 +1,279 @@
+  // Reference map from (thr,val) to bit
+  using RefLayout = SrcLayout;
+
+  GTensor tensor;
+
+  template <class TS, class SLayout, class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void copy_unpack(
+      Copy_Traits const &traits,
+      Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+      Tensor<TD, DLayout> &dst) {
+    static_assert(is_rmem<TD>::value);
+    int H = size<0>(traits.tensor);
+    // int W = size<1>(traits.tensor) * sizeof(typename
+    // decltype(traits.tensor)::engine_type::value_type);
+    int W =
+        size<1>(traits.tensor) *
+        sizeof(typename TD::value_type); // TODO: inconsistent to give the size
+                                         // in elements but use vector for copy
+    auto [y, x] = src.data().coord_;
+    XE_2D_LOAD::copy(traits.tensor.data().get(), W, H, W, int2_{x, y},
+                     &*dst.data());
+  }
+
+  template <class TS, class SLayout, class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void
+  copy_unpack(Copy_Traits const &traits, Tensor<TS, SLayout> const &src,
+              Tensor<ViewEngine<ArithmeticTupleIterator<TD>>, DLayout> &dst) {
+    static_assert(is_rmem<TS>::value);
+    int H = size<0>(traits.tensor);
+    int W = size<1>(traits.tensor) *
+            sizeof(typename decltype(traits.tensor)::engine_type::value_type);
+    auto [y, x] = dst.data().coord_;
+    XE_2D_SAVE::copy(traits.tensor.data().get(), W, H, W, int2_{x, y},
+                     &*src.data());
+  }
+};
+
+template <class GTensor> struct Copy_Traits<XE_2D_U16X4X2_N, GTensor> {
+  // using ThrID   = Layout<_16>; //TODO: I think it should be 16 (copy is per
+  // subgroup) - but static_assert fails
+  using ThrID = Layout<_16>;
+  // Map from (src-thr,src-val) to bit
+  using SrcLayout = Layout<Shape<_16, _64>, Stride<_0, _1>>;
+  // Map from (dst-thr,dst-val) to bit
+  using DstLayout =
+      Layout<Shape<_16, Shape<Shape<_8, _4>, Shape<_16, _2>>>,
+             Stride<_16, Stride<Stride<_512, _4096>, Stride<_1, _256>>>>;
+  // Reference map from (thr,val) to bit
+  using RefLayout = DstLayout;
+
+  GTensor tensor;
+
+  template <class TS, class SLayout, class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void copy_unpack(
+      Copy_Traits const &traits,
+      Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+      Tensor<TD, DLayout> &dst) {
+    static_assert(is_rmem<TD>::value);
+    int H = size<0>(traits.tensor);
+    // int W = size<1>(traits.tensor) * sizeof(typename
+    // decltype(traits.tensor)::engine_type::value_type);
+    int W =
+        size<1>(traits.tensor) *
+        sizeof(typename TD::value_type); // TODO: inconsistent to give the size
+                                         // in elements but use vector for copy
+    auto [y, x] = src.data().coord_;
+    XE_2D_U16X4X2_N::copy(traits.tensor.data().get(), W, H, W, int2_{x, y},
+                          &*dst.data());
+  }
+};
```

Comment:
All these copy unpack functions look the same apart from the 'XE_2D_U16X4X2_N' part. Could it be possible to generalise this the same way NVIDIA does? We should try to avoid having to implement all of this for each new copy operation.